### PR TITLE
feat: git workflow

### PR DESCRIPTION
**Pull request by mikevic**

### CODE REVIEW

File: domains/engineering/secure/rails_secure_credentials.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	We can use more files based on the environment
+
+	Examples
+
+		config/credentials/production.key
+		config/credentials/staging.key
+		config/credentials/development.key
+
+2. But you should save your key which you added inside the file `config/credentials/production.key` somewhere else because this is the key used to encrypt the credentials, so if we miss this key then we can't decrypt our file to edit or use the credentials inside our `config/credentials/production.yml.enc`   
+3. Then we need to keep the file `config/credentials/production.key` as secret so it should be added under `.gitignore`
```
Comment (camiloforero) [inline domains/engineering/secure/rails_secure_credentials.md]: Is this step needed? when the command `EDITOR="subl --wait" rails credentials:edit --environment ENV` is run, the ENV.key file is automatically added to the .gitignore

File: domains/engineering/secure/rails_secure_credentials.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+title: Rails secure credentials
+nav_order: 5
+parent: Secure
+grand_parent: Engineering
+---
+In the older version of rails 5.2 we place all our credentials inside  `config/secrets.yml` . But later rails 5.2 and upcoming versions replaced the current config/secrets.yml into config/secrets.yml.enc
+
+**Before Using Rails Credentials**
+
+1. So before using rails credentials the main thing to remember is to add a key inside `config/credentials/production.key`
```
Comment (camiloforero) [inline domains/engineering/secure/rails_secure_credentials.md]: Style: Restructure your sentences so each step of an instruction lists starts with a verb. In this case, "Add a key inside `config/credentials/production.key` before using rails credentials".

File: domains/engineering/secure/rails_secure_credentials.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	We can use more files based on the environment
+
+	Examples
+
+		config/credentials/production.key
+		config/credentials/staging.key
+		config/credentials/development.key
+
+2. But you should save your key which you added inside the file `config/credentials/production.key` somewhere else because this is the key used to encrypt the credentials, so if we miss this key then we can't decrypt our file to edit or use the credentials inside our `config/credentials/production.yml.enc`
```
Comment (camiloforero) [inline domains/engineering/secure/rails_secure_credentials.md]: Style: "Save the key that you added somewhere else: This is the key used to encrypt..."
Keep the first statement as a short sentence starting with a verb. THen if you have extra information. put it in a new paragraph.

File: domains/engineering/secure/rails_secure_credentials.md (hunk 1 above)
Comment (mikevic) [reply domains/engineering/secure/rails_secure_credentials.md]: @camiloforero - yes, this is required when you have different files for each environment. You can the --environment when you want to edit the credentials for a particular environment.

Comment (camiloforero) [reply domains/engineering/secure/rails_secure_credentials.md]: What I mean is, whenever I run that command, an extra line to the .gitignore is automatically added at the end, so I don't need to do that manually

Comment (mikevic) [reply domains/engineering/secure/rails_secure_credentials.md]: ahhh yes correct! @BALAJI-BALA - we can remove that out!

**Pull request merged**